### DISCUSSION TITLE
smooth-scrollbar + vue, nuxt and gsap

### DISCUSSION THREAD

**Discussion by cyprianwaclaw** [idiotWu/smooth-scrollbar#497] [Show and tell]
How to use smooth-scrollbar with nuxt3 and gsap ?

**Comment by sadeghbarati** [comment]
Hi @cyprianwaclaw 

I'm working on **Vue** & **Nuxt**

The package is using `vue-demi` so it's compatible with 

- nuxt v2.15.* -  nuxt bridge -  nuxt v3-rc
- vue v2.6.14  -   vue 2.7.*     -  vue v3.2

<del>But `nuxt-module` still needs to be done, just for `autoImport` feature nuxt provides</del> probably add custom preset for [unplugin-auto-import](https://github.com/antfu/unplugin-auto-import)
Also it's can be used with [gsap and ScrollTrigger](https://greensock.com/docs/v3/Plugins/ScrollTrigger/static.scrollerProxy())


https://nuxt2-15-8-smooth-scrollbar.vercel.app/

I'll include the source code of this demo

**Reply by sadeghbarati** [reply]
Btw website loading is so **slow** because of [TailwindCSS cdn source code](https://cdn.jsdelivr.net/npm/tailwindcss@2.1.2/dist/tailwind.min.css) 

I'll work on better demo

**Reply by cyprianwaclaw** [reply]
Hi, thank you so much, could you send me link repo vue-demi ?

**Reply by sadeghbarati** [reply]
It's in organization private repo, I'll share in public repo as soon as I test Vue 2.7 completely

**Comment by sadeghbarati** [comment]
https://www.npmjs.com/package/@smooth-scrollbar-contrib/vue-test 


#### Usually this plugin can be used per page or in `layouts` (app-wide)
you can refer to the source or README on npm for now, three ways to use smooth-scrollbar

1. Component
2. Composable => you can also init Scrollbar on `__nuxt` element too
3. Directive

use Vue component if you want to use it with `ScrollTrigger`, component has the **`mounted`** emit

- `@mounted` => ({ target, scrollbarIns}) - also you can use this for register [scrollerProxy](https://greensock.com/docs/v3/Plugins/ScrollTrigger/static.scrollerProxy())

> **Warning**: if you are using component to use ScrollTrigger animations you have to use `nextTick` (wait for all rendering including rendering Scrollbar component) further reading in https://next-blog.site/media/nuxtscrolltrigger/


**`layouts/default.vue`:**
```vue
<script lang="ts" setup>
import gsap from 'gsap'
import ScrollTrigger from 'gsap/ScrollTrigger'
import SmoothScrollbar from 'smooth-scrollbar';
import OverscrollPlugin from 'smooth-scrollbar/plugins/overscroll'
import { Scrollbar, type ScrollbarOptions, type MountedEmitPayload } from '@smooth-scrollbar-contrib/vue-test'

function mountedFn({target, scrollbarIns}: MountedEmitPayload) {
  ScrollTrigger.scrollerProxy(target.value, {
    scrollTop(value) {
      if (arguments.length) {
        scrollbarIns.scrollTop = value;
      }
      return scrollbarIns.scrollTop;
    }
  });

  scrollbarIns.addListener(ScrollTrigger.update);
  ScrollTrigger.defaults({ scroller: target.value });
}
</script>

<template>
  <Scrollbar @mounted="mountedFn" scrollbar-options="">
    <slot></slot>
  </Scrollbar >
</template>
```
